### PR TITLE
Add gcc arm embedded toolchain [BUILD-651]

### PR DESCRIPTION
**Pull request by krisukox**
This PR adds gcc arm embedded toolchain.

# Testing
https://github.com/swift-nav/starling/pull/7844

### CODE REVIEW

File: cc/toolchains/gcc_arm_embedded/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+    srcs = [
+        ":compiler_files",
+        ":linker_files",
+        "@gcc_arm_embedded_toolchain//:bin",
+    ],
+    tags = ["manual"],
+)
+
+constraint_value(
+    name = "cortex-m4",
```
Comment (isaactorz) [inline cc/toolchains/gcc_arm_embedded/BUILD.bazel:82]: Move these to `cc/constraints/BUILD.bazel` ?

File: cc/toolchains/gcc_arm_embedded/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+platform(
+    name = "cortex_m7_none_eabi",
+    constraint_values = [
+        "@platforms//os:none",
+        ":cortex-m7",
+    ],
+)
+
+platform(
+    name = "cortex_m3_none_eabi",
```
Comment (isaactorz) [inline cc/toolchains/gcc_arm_embedded/BUILD.bazel:140]: should we move these to the `platforms/` folder?

**Review by isaactorz — COMMENTED** [review]
Two minor suggestions about file locations, otherwise 👍🏼

File: cc/toolchains/gcc_arm_embedded/BUILD.bazel (hunk 2 above)
Comment (krisukox) [reply cc/toolchains/gcc_arm_embedded/BUILD.bazel]: Done

**Review by isaactorz — APPROVED** [review]

**Review by isaactorz — APPROVED** [review]

File: cc/toolchains/gcc_arm_embedded/init.bzl [hunk 3]
```diff
@@ -0,0 +1,10 @@
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+def gcc_arm_embedded_toolchain():
```
Comment (isaactorz) [inline cc/toolchains/gcc_arm_embedded/init.bzl:3]: We should add this function to `cc/respositories.bzl` instead of creating this file.

**Review by isaactorz — CHANGES_REQUESTED** [review]
Sorry I realized there's one small change I would like to see before we merge this.

**Review by isaactorz — APPROVED** [review]

**Comment by isaactorz** [comment]
I'm going to go ahead and merge this one.

**Pull request merged**